### PR TITLE
add bounty to rebase

### PR DESCRIPTION
**Pull request by olyzeus**
recompensate user, or compensate keeper, for gas cost of rebase() function.

### CODE REVIEW

**Review by ind-igo — APPROVED** [review]

File: contracts/Staking.sol [hunk 1]
```diff
@@ -91,10 +91,10 @@ contract OlympusStaking is OlympusAccessControlled {
         bool _rebasing,
         bool _claim
     ) external returns (uint256) {
-        rebase();
-
+        if (rebase() && address(distributor) != address(0)) {
```
[the remark this answers is not in the document]
Comment (olyzeus) [reply contracts/Staking.sol:94]: addressed

File: contracts/Staking.sol [hunk 2]
```diff
@@ -175,18 +175,19 @@ contract OlympusStaking is OlympusAccessControlled {
         bool _trigger,
         bool _rebasing
     ) external returns (uint256 amount_) {
+        amount_ = _amount;
+        uint256 bounty;
         if (_trigger) {
-            rebase();
+            if (rebase() && address(distributor) != address(0)) {
```
[the remark this answers is not in the document]
Comment (olyzeus) [reply contracts/Staking.sol:181]: addressed

File: contracts/Staking.sol [hunk 3]
```diff
@@ -211,29 +211,31 @@ contract OlympusStaking is OlympusAccessControlled {
...
 
-            if (distributor != address(0)) {
-                IDistributor(distributor).distribute();
+            uint256 bounty;
+            if (address(distributor) != address(0)) {
+                bounty = distributor.distribute();
             }
 
-            if (contractBalance() <= totalStaked()) {
+            if (contractBalance() <= totalStaked().add(bounty)) {
```
[the remark this answers is not in the document]
Comment (olyzeus) [reply contracts/Staking.sol:236]: addressed. contractBalance() and totalStaked() removed, replaced by local vars querying their values

File: contracts/StakingDistributor.sol [hunk 4]
```diff
@@ -62,16 +63,20 @@ contract Distributor is IDistributor, OlympusAccessControlled {
...
+    function distribute() external override returns (uint256) {
         require(msg.sender == staking, "Only staking");
-
         // distribute rewards to each recipient
         for (uint256 i = 0; i < info.length; i++) {
             if (info[i].rate > 0) {
-                treasury.mint(info[i].recipient, nextRewardAt(info[i].rate)); // mint and send from treasury
+                if (info[i].recipient == staking && bounty != 0) {
+                    treasury.mint(info[i].recipient, nextRewardAt(info[i].rate).add(bounty)); // mint and send with bounty
+                } else {
```
[the remark this answers is not in the document]
Comment (olyzeus) [reply contracts/StakingDistributor.sol:73]: bounty check removed. struct addition should be queued for future update (distributor can be replaced)

**Review by ind-igo — CHANGES_REQUESTED** [review]

**Review by ind-igo — APPROVED** [review]

**Pull request merged**